Example/GiniVision/ComponentAPICoordinator.swift

```diff
@@ -168,151 +206,323 @@ final class ComponentAPICoordinator: NSObject, Coordinator {
+                guard let `self` = self else { return }
+                switch result {
+                case .success(let extractions):
+                    self.handleAnalysis(with: extractions)
+                case .failure(let error):
+                    break
+                }
+            }
+        })
+    }
+    
+    fileprivate func delete(document: GiniVisionDocument) {
+        documentService?.delete(document)
+    }
+}
+
+// MARK: - Other
+
+extension ComponentAPICoordinator {
+    
+    fileprivate func setupTabBar() {
+        let navTabBarItem = UITabBarItem(title: "New document", image: UIImage(named: "tabBarIconNewDocument"), tag: 0)
+        let helpTabBarItem = UITabBarItem(title: "Help", image: UIImage(named: "tabBarIconHelp"), tag: 1)
+        
+        self.navigationController.tabBarItem = navTabBarItem
+        self.componentAPIOnboardingViewController.tabBarItem = helpTabBarItem
+        
+        self.componentAPITabBarController.setViewControllers([navigationController,
+                                                              componentAPIOnboardingViewController],
+                                                             animated: true)
+    }
+    
+    fileprivate func addCloseButtonIfNeeded(onViewController viewController: UIViewController) {
+        if navigationController.viewControllers.isEmpty {
+            viewController.navigationItem.leftBarButtonItem = UIBarButtonItem(title: "Schließen",
+                                                                              style: .plain,
+                                                                              target: self,
+                                                                              action: #selector(closeComponentAPI))
+        }
     }
     
     func didTapRetry() {
-        if (newDocumentViewController.viewControllers.compactMap { $0 as? ComponentAPICameraViewController}).first == nil {
+        if (navigationController.viewControllers.compactMap { $0 as? CameraViewController}).first == nil {
             closeComponentAPI()
             return
         }
 
-        newDocumentViewController.popToRootViewController(animated: true)
+        navigationController.popToRootViewController(animated: true)
     }
 }
 
 // MARK: UINavigationControllerDelegate
 
 extension ComponentAPICoordinator: UINavigationControllerDelegate {
+    
     func navigationController(_ navigationController: UINavigationController,
-                              willShow viewController: UIViewController,
-                              animated: Bool) {
-        guard let fromViewController = navigationController
-            .transitionCoordinator?.viewController(forKey: .from) else { return }
-        newDocumentViewController
-            .setNavigationBarHidden(viewController is ComponentAPICameraViewController, animated: true)
+                              animationControllerFor operation: UINavigationControllerOperation,
```

Comment:
I cannot agree with this method. The whole extension is supposed to be a `UINavigationControllerDelegate` and this method is supposed to handle transitions, but there is a a whole lot of business logic. In particular deleting documents, cancelling analysis and sending feedback.

Not to mention that it's too long and does too many things.